src/main/java/com/powsybl/openloadflow/sensi/DcSensitivityAnalysis.java

```diff
@@ -713,22 +905,33 @@ private void detectConnectivityLoss(LfNetwork lfNetwork, DenseMatrix states, Lis
                 ));
             }
             // compute the contingencies with loss of connectivity
-            for (Map.Entry<List<Set<ComputedContingencyElement>>, List<Contingency>> entry : contingenciesByGroupOfElementsBreakingConnectivity.entrySet()) {
-                List<Set<ComputedContingencyElement>> groupOfElementsBreakingConnectivity = entry.getKey();
+            for (Map.Entry<Set<ComputedContingencyElement>, List<Contingency>> entry : contingenciesByGroupOfElementsBreakingConnectivity.entrySet()) {
+                Set<ComputedContingencyElement> breakingConnectivityCandidates = entry.getKey();
                 List<Contingency> contingencyList = entry.getValue();
 
                 lfFactors.forEach(factor -> factor.setPredefinedResult(null));
-                groupOfElementsBreakingConnectivity.stream().flatMap(Set::stream)
-                        .map(ComputedContingencyElement::getElement)
-                        .map(element -> lfNetwork.getBranchById(element.getId()))
-                        .forEach(lfBranch -> connectivity.getConnectivity().cut(lfBranch.getBus1(), lfBranch.getBus2()));
+                cutConnectivity(lfNetwork, connectivity.getConnectivity(), breakingConnectivityCandidates);
+
+                Integer slackBusComponent = connectivity.getConnectivity().getComponentNumber(lfNetwork.getSlackBus());
+
+                for (LfSensitivityFactor factor : lfFactors) {
+                    // check if the factor function and variable are in different connected components
+                    if (factor.areVariableAndFunctionDisconnected(connectivity.getConnectivity())) {
+                        factor.setPredefinedResult(0d);
+                    } else if (!factor.isConnectedToSlack(slackBusComponent, connectivity.getConnectivity())) {
+                        factor.setPredefinedResult(Double.NaN);
+                    }
+                }
+
+                rescaleGlsk(factorGroups, connectivity.getConnectivity(), slackBusComponent);
```

Comment:
@annetill here, the call to this function changes the variable glskMapInMainComponent from the glsk factor group, and then, it will change the output of setInjectionByBus when it is called by computeInjectionFactors line 951